src/components/PasswordToggle/PasswordToggle.tsx

```diff
@@ -0,0 +1,135 @@
+import classNames from "classnames";
+import React, { useEffect, useRef, useState } from "react";
+import type { InputHTMLAttributes, ReactNode } from "react";
+
+import Button from "../Button";
+import Field from "../Field";
+import Label from "../Label";
+
+import type { ClassName, PropsWithSpread } from "types";
+
+/**
+ * The props for the Input component.
+ */
+export type Props = PropsWithSpread<
+  {
+    /**
+     * The content for caution validation.
+     */
+    caution?: ReactNode;
+    /**
+     * Optional class(es) to pass to the input element.
+     */
+    className?: ClassName;
+    /**
+     * The content for error validation.
+     */
+    error?: ReactNode;
+    /**
+     * Help text to show below the field.
+     */
+    help?: ReactNode;
+    /**
+     * The id of the input.
+     */
+    id: string;
+    /**
+     * The label for the field.
+     */
+    label?: ReactNode;
+    /**
+     * Whether the field is read only.
+     */
+    readOnly?: boolean;
+    /**
+     * Whether the field is required.
+     */
+    required?: boolean;
+    /**
+     * The content for success validation.
+     */
+    success?: ReactNode;
+    /**
+     * Whether to focus on the input on initial render.
+     */
+    takeFocus?: boolean;
+    /**
+     * Optional class(es) to pass to the wrapping Field component
+     */
+    wrapperClassName?: string;
+  },
+  InputHTMLAttributes<HTMLInputElement>
+>;
+
+const PasswordToggle = ({
+  caution,
+  className,
+  error,
+  help,
+  id,
+  label,
+  readOnly,
+  required,
+  success,
+  takeFocus,
+  wrapperClassName,
+  ...inputProps
```

Comment:
We've had issues with this spread in the past, can you add a test that input props get passed to the input itself.